### ISSUE TITLE
Unable to export database from the extension

### ISSUE THREAD

**Issue by hajekj** [shibayan/phpMyAdminExtension#1]
When trying to export database, the server returns HTTP 403 code and status page instead of the database export.

**Comment by shibayan** [comment]
@hajekj I tried exporting by choosing SQL for the format and it succeeded.
If PHP is caused, an error log may be output to "D:\site\LogFiles" dir.

**Comment by hajekj** [comment]
Unfortunately, there are no logs. It simply returns 403 forbidden error.

![capture](https://cloud.githubusercontent.com/assets/8337913/23587847/6f3ce6d8-01b4-11e7-956f-3da4e645b252.PNG)

**Comment by shibayan** [comment]
This error is related to Web App's quota.

Refer to the following URL.
https://blogs.msdn.microsoft.com/waws/2016/01/05/azure-web-apps-error-403-this-web-app-is-stopped/

**Comment by hajekj** [comment]
This is weird since it is running in a Standard (S1) plan.

**Comment by shibayan** [comment]
Umm. It is certainly weird.
But this screen can be displayed only on the runtime of App Service, I do not think it is a problem of phpmyadmin.

In the case of quota problems, information should remain in Azure Portal metrics.

**Comment by hajekj** [comment]
Any guidance on how to debug this? This hasn't happened before, I can use the PMA without any issue..

**Comment by davidebbo** [comment]
Is 'hajekj-pma' the Web App name (based on screenshot)? At this time, this app does not appear to exist. Did you delete it?

**Comment by hajekj** [comment]
I deleted it, but I can re-create it.

**Comment by hajekj** [comment]
I re-created it, still showing the error.

**Comment by davidebbo** [comment]
I'll take a look.

**Comment by davidebbo** [comment]
And you get this error just by browsing to https://hajekj-pma.scm.azurewebsites.net/?

It's odd, as the site seems to be running normally from what I can see.

**Comment by hajekj** [comment]
https://hajekj-pma.scm.azurewebsites.net/phpmyadmin when trying to export the database.

**Comment by davidebbo** [comment]
Can you try just going to https://hajekj-pma.scm.azurewebsites.net/? I'm guessing you'll see the same, and that will help us isolate that it's unrelated to phpmyadmin.

**Comment by hajekj** [comment]
I can access SCM and phpMyAdmin, I just get the error when trying to export a database.

**Comment by hajekj** [comment]
https://gyazo.com/5606aad04fd7562b48f9bb500a7c8aff

**Comment by davidebbo** [comment]
Ah, thanks, that helps! I'll take a further look.

**Comment by davidebbo** [comment]
@hajekj Do you mind if I try something on the site as part of investigating?

**Comment by davidebbo** [comment]
@suwatch investigated. Root issue is that PhpMyAdmin is doing an adhoc FORM POST from JS and not passing Referer, which is not supported with scm site authentication.

We'll probably need to get that patched in PhpMyAdmin.

**Comment by hajekj** [comment]
@davidebbo Sure, try whatever you need to.

**Comment by shibayan** [comment]
It seems that I did not reproduce it in my environment because I used Microsoft Edge.
With Chrome it was reproduced on the login screen. I might be able to write a patch :)

**Comment by suwatch** [comment]
@shibayan users reported on both Chrome and Firefox.   Yes, it works on Edge.   Once you get a chance to patch it, share us the commit id.

**Comment by rooby** [comment]
Another workaround I was able to use (I don't have access to IE right now) was mysqldump via powershell.

**Comment by suwatch** [comment]
Another workaround to consider is, if you are using [In-App MySql feature](https://blogs.msdn.microsoft.com/appserviceteam/2017/03/06/announcing-general-availability-for-mysql-in-app/), you could simply browse to `https://<site>.scm.azurewebsites.net/mysqlutils` and follow the instruction how to export or import mysql database.   If you are using external my sql (such as cleardb), you will need to turn on mysqlutils explicitly by adding appSettings `MYSQLUTILS_EXTENSION_VERSION = latest`.   Hope this helps,

**Comment by shibayan** [comment]
@suwatch The cause was found out. Everything was in the Referrer-Policy setting of phpMyAdmin.
http://www.w3.org/TR/referrer-policy/

phpMyAdmin specified no-referrer as policy. Edge does not support policies, so it seems there was no effect. The nupkg creation script now converts "no-referrer" to "origin".
https://github.com/shibayan/phpMyAdminExtension/blob/master/download.ps1#L15

It now works with Chrome :)

**Comment by suwatch** [comment]
Thanks for the investigation.  We will patch the fix ETA for rollout 1-2 weeks.

**Comment by davidebbo** [comment]
I think I need to clear some possible confusion here:

- @shibayan owns the phpMyAdmin Site Extension in the gallery: http://www.siteextensions.net/packages/phpmyadmin/. He just updated it and the new version is available right now (and not in 1-2 weeks)
- Separately from that, we now have a pre-installed copy of phpMyAdmin on the VMs, and that's what @suwatch is referring to. That one is enabled not from the gallery, but by setting `PHPMYADMIN_EXTENSION_VERSION = latest` in the Azure App Settings. This is the one that will take 1-2 weeks to update, once @suwatch makes the same fix as @shibayan in there.

**Comment by shibayan** [comment]
@davidebbo @suwatch I agree. I thought that it is difficult for users to understand that there are two phpMyAdmin.
I think that the difference with pre-installed phpMyAdmin is about In-App MySQL support, so I think that it is one thing to merge into either.

Personal opinion, since phpMyAdmin has many updates, I think that it is advantageous to provide it as a site extension. Ideal to manage with github.com/projectkudu.

**Comment by ksbecker** [comment]
I know this is an old issue, but I'm facing this today. I don't know if it's the same problem, but trying to export from phpMyAdmin I'm getting a 403 error. I looked in the header.php file and the Referrer-Policy was set to no-referrer. I changed that to origin and that didn't seem to help. I did not restart the server yet; not sure if that's necessary.

I'm running 4.7.9 as an extension installed via the Azure Portal.

Any suggestions?

**Comment by josegomera** [comment]
> I know this is an old issue, but I'm facing this today. I don't know if it's the same problem, but trying to export from phpMyAdmin I'm getting a 403 error. I looked in the header.php file and the Referrer-Policy was set to no-referrer. I changed that to origin and that didn't seem to help. I did not restart the server yet; not sure if that's necessary.
> 
> I'm running 4.7.9 as an extension installed via the Azure Portal.
> 
> Any suggestions?

Hi, I'm facing the same issue too. Same version as @ksbecker

**Comment by michaelmowry** [comment]
Same error here and it even occurs using MS Edge. @shibayan do you have another patch that can be applied?

**Comment by mkabab** [comment]
the same issue

**Comment by gal-samuel-israel** [comment]
I Had the same 403 response on DB export after installing 4.7.9 to Azure MySql in app instead of the built-in 4.6.6.
I used the Kudu Debug console to edit the file:  D:\home\SiteExtensions\phpmyadmin\libraries\Header.php
and replace all occurrences of referrer related headers from "no-origin" to "origin".
Then I restarted the instance and the problem was solved.
